### PR TITLE
Emmanuel/dao/reservation

### PR DESCRIPTION
**Pull request by emmanuel099**

### CODE REVIEW

File: src/main/java/com/at/ac/tuwien/sepm/ss15/edulium/dao/impl/DBReservationDAO.java [hunk 1]
```diff
@@ -0,0 +1,501 @@
...
+            } else {
+                LOGGER.error("retrieving the change number failed");
+                throw new DAOException("retrieving the change number failed");
+            }
+        } catch (SQLException e) {
+            LOGGER.error("generating history for reservation failed", e);
+            throw new DAOException("generating history for reservation failed", e);
+        }
+
+        assert changeNr >= 0;
```
Comment (PhilippParis) [inline src/main/java/com/at/ac/tuwien/sepm/ss15/edulium/dao/impl/DBReservationDAO.java]: changeNr can never be <= 0

File: src/main/java/com/at/ac/tuwien/sepm/ss15/edulium/dao/impl/DBReservationDAO.java [hunk 2]
```diff
@@ -0,0 +1,501 @@
...
+        return tables;
+    }
+
+    /**
+     * searches all tables from the reservation-table associations for the given reservation history with the given changeNumber
+     * @param reservation reservation history dataset
+     * @param changeNumber reservation history change number
+     * @throws DAOException if an error accessing the database occurred
+     */
+    private List<Table> getTablesForHistoryReservation(Reservation reservation, long changeNumber) throws DAOException {
```
Comment (PhilippParis) [inline src/main/java/com/at/ac/tuwien/sepm/ss15/edulium/dao/impl/DBReservationDAO.java]: "getTablesForReservationHistory"?

File: src/main/java/com/at/ac/tuwien/sepm/ss15/edulium/dao/impl/DBReservationDAO.java (hunk 1 above)
Comment (emmanuel099) [inline src/main/java/com/at/ac/tuwien/sepm/ss15/edulium/dao/impl/DBReservationDAO.java]: Should I remove the assert in your opinion or correct it to >= 1 (had >= 1 first but then I wanted to change it to > 0 but missed the = somehow :/)?

The assert is here to express, that we have a valid changeNr from this lines on.

File: src/test/java/com/at/ac/tuwien/sepm/ss15/edulium/dao/TestReservationDAO.java [hunk 3]
```diff
@@ -0,0 +1,914 @@
...
+        assertNotNull(reservation.getIdentity());
+
+        // try to find the reservation and compare it
+        List<Reservation> storedObjects = reservationDAO.find(Reservation.withIdentity(reservation.getIdentity()));
+        assertEquals(1, storedObjects.size());
+        assertEquals(reservation, storedObjects.get(0));
+    }
+
+    @Test(expected = ValidationException.class)
+    public void testCreate_addingInvalidObjecShouldFail() throws DAOException, ValidationException {
```
Comment (PhilippParis) [inline src/test/java/com/at/ac/tuwien/sepm/ss15/edulium/dao/TestReservationDAO.java]: addingInvalidObjec>t<ShouldFail

**Pull request merged**